### PR TITLE
Build ykrustc instead of using a binary snapshot.

### PR DESCRIPTION
**Pull request by vext01**
This means that we can detect changes in yk which would break the compiler build. A couple instances of that happening have arisen recently.

### CODE REVIEW

File: .buildbot.sh [hunk 1]
```diff
@@ -2,27 +2,15 @@
 
 set -e
 
+YKRUSTC_REPO=https://github.com/softdevteam/ykrustc
```
Comment (ltratt) [inline .buildbot.sh:5]: I must admit, this makes the script harder to read IMHO: it's only used in one place, and the code that comes after it heavily depends on the repo being called `ykrustc`. I'd suggest inlining this variable TBH.

File: .buildbot.sh [hunk 2]
```diff
@@ -33,5 +21,27 @@ sh rustup.sh --default-host x86_64-unknown-linux-gnu \
...
+rustup override set ykrustc-stage1
+
+# Test both workspaces using the compiler we just built.
+export RUSTFLAGS="-C tracer=${CI_TRACER_KIND} -D warnings"
+cargo xtask test
+cargo xtask bench
+cargo xtask clean
+
+# Also test the build without xtask, as that's what consumers will do.
+cargo build
```
Comment (ltratt) [inline .buildbot.sh:41]: Does this work -- I thought we *had* to use `xtask`?

File: .buildbot.sh [hunk 3]
```diff
@@ -33,5 +21,27 @@ sh rustup.sh --default-host x86_64-unknown-linux-gnu \
...
+
+# Run rustfmt.
```
Comment (ltratt) [inline .buildbot.sh:43]: Given how lengthy the preceding stages are, should we do `fmt` first? We also won't have to unset `RUSTFLAGS` that way!

File: .buildbot.sh (hunk 2 above)
Comment (bjorn3) [reply .buildbot.sh:41]: Yes it does work for `cargo build`, but all warnings and build progress for the internal workspace is hidden as build script output is only shown when it fails.

Comment (vext01) [reply .buildbot.sh:41]: We tested the xtask case a few lines above. This is testing what crates consuming us will do (we have no way to have consumers run xtask, so we had to duplicate some code in `build.rs` -- it's complicated).

File: .buildbot.sh (hunk 3 above)
Comment (vext01) [reply .buildbot.sh:43]: I wondered that myself. But can we do it as a separate PR, as testing these things properly it time consuming.

Comment (ltratt) [reply .buildbot.sh:43]: TBH, I'm happy with you raising the commit and doing `bors try`1

**Comment by vext01** [comment]
bors try

**Comment by vext01** [comment]
bors try

**Comment by vext01** [comment]
OK to squash?

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Comment by ltratt** [comment]
Sorry, I confused myself as to which PR is which! This one is fine.

bors r+

**Pull request merged**